xapian-letor/tests/api_letor.cc

```diff
@@ -0,0 +1,187 @@
+
+    return true;
+}
+
+static void
+gen_uniqterms_gt_doclen_db(Xapian::WritableDatabase& db, const string&)
+{
+    Xapian::Document doc;
+    Xapian::TermGenerator termgenerator;
+    termgenerator.set_document(doc);
+    termgenerator.set_stemmer(Xapian::Stem("en"));
+    termgenerator.index_text("Lions, Tigers, Bears and Giraffes", 1, "S");
+    termgenerator.index_text("This paragraph talks about lions and tigers and "
+			     "bears (oh, my!). It mentions giraffes, "
+			     "but that's not really very important. Lions "
+			     "and tigers are big cats, so they must be really "
+			     "cuddly. Bears are famous for being cuddly, at "
+			     "least when they're teddy bears.", 1, "XD");
+    termgenerator.index_text("Lions, Tigers, Bears and Giraffes");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("This paragraph talks about lions and tigers and "
+			     "bears (oh, my!). It mentions giraffes, "
+			     "but that's not really very important. Lions "
+			     "and tigers are big cats, so they must be really "
+			     "cuddly. Bears are famous for being cuddly, at "
+			     "least when they're teddy bears.");
+    db.add_document(doc);
+    doc.clear_terms();
+    termgenerator.index_text("Lions, Tigers and Bears", 1, "S");
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.", 1,
+			     "XD");
+    termgenerator.index_text("Lions, Tigers and Bears");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("This is the paragraph of interest. Tigers are "
+			     "massive beasts - I wouldn't want to meet a "
+			     "hungry one anywhere. Lions are scary even when "
+			     "lyin' down. Bears are scary even when bare. "
+			     "Together I suspect they'd be less scary, as the "
+			     "tigers, lions, and bears would all keep each "
+			     "other busy. On the other hand, bears don't live "
+			     "in the same continent as far as I know.");
+    db.add_document(doc);
+}
+
+DEFINE_TESTCASE(ranker1, generated) {
+    Xapian::Ranker *ranker = new Xapian::ListNETRanker();
+    string a = get_database_path("apitest_ranker1",
+				       gen_uniqterms_gt_doclen_db);
+    Xapian::Enquire enquire((Xapian::Database(a)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    string query ="20001 'lions'";
+    string qrel = "20001 Q0 1 1\n20001 Q0 2 2";
+    std::ofstream myfile;
+    myfile.open("query.txt");
+    myfile << query;
+    myfile.close();
+    myfile.open("qrel.txt");
+    myfile << qrel;
+    myfile.close();
```

Comment:
You're using the exact same filenames for dynamically created files in multiple testcases, which is unnecessarily hostile to running testcases in parallel.

But the contents of these files are known in advance, so why not just add them to the source tree?  That'll be faster than creating them over and over, and reduce the source code size by more than the size of a single copy of the text in a file.